llvm/test/Transforms/LoopUnroll/AMDGPU/unroll-cost-addrspacecast.ll

```diff
@@ -50,7 +50,7 @@ for.end:
 }
 
 ; CHECK-LABEL: @test_func_addrspacecast_cost_nonfree(
-; CHECK: br i1 %exitcond
+; CHECK-NOT: br i1 %exitcond
```

Comment:
I think you should adjust the `-unroll-threshold=49` option in this test to preserve the old behavior. If I understood correctly, the extra cost saving applies to all functions in this file equally, so just adjusting the threshold should work.

Reply:
Unfortunately, with this change it's now impossbile to find threshold that keeps old behavour for all tests in this file. The savings in test_func_addrspacecast_cost_nonfree are higher (I assume that for AMDGPU target GEP with constant offset is cheaper in addrspace(3)), so there is no threshold where first 2 tests are unrolled and the last is not.

Reply:
@arsenm Any idea how to preserve the intent of this test?

Reply:
I don't know have any great ideas. Either split the test file up, or add some filler operations to up the costs in other functions

Reply:
I've come up with the following solution: add some extra level of indirection to get GEP indices so they will not become constant after loop unrolling; this hides the changes introduced by this patch and preserves the behaviour of original test (with slight increasing of unroll threshold). This is done by loading the GEP index from some global array (the additional cost of this loading is the same for all tests).